providers/hashicorp/src/airflow/providers/hashicorp/_internal_client/vault_client.py

```diff
@@ -299,13 +306,47 @@ def _auth_gcp(self, _client: hvac.Client) -> None:
         )
 
         scopes = _get_scopes(self.gcp_scopes)
-        credentials, _ = get_credentials_and_project_id(
+        credentials, project_id = get_credentials_and_project_id(
             key_path=self.gcp_key_path, keyfile_dict=self.gcp_keyfile_dict, scopes=scopes
         )
+
+        import time
+        import json
+        import googleapiclient
+
+        with open(self.gcp_key_path, 'r') as f:
```

Comment:
This will work only with a mounted credentials file. We also have to handle cases when Airflow is running using workload identity (or Composer).

Maybe we can reuse gcloud SDK to create a signed JWT?